files/image_config/monit/arp_update_checker

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+"""
+arp_update_checker
+
+This script is intended to be run by Monit. It will write an alerting message into
+syslog if it found arp_update script stuck on ping command.
+Then it will attempt to restart arp_update script.
+
+The following is an example in Monit configuration file to show how Monit will run
+this script:
+
+check program arp_update_checker with path "/usr/bin/arp_update_checker" every 10 cycles
+    if status != 0 for 3 times within 3 cycles then alert repeat every 1 cycles
+"""
+
+import syslog
+import subprocess
+import sys
+
+TIMEOUT=5.0
+
+def log_info(message):
+    syslog.syslog(syslog.LOG_INFO, message)
+
+def log_warning(message):
+    syslog.syslog(syslog.LOG_WARNING, message)
+
+def log_error(message):
+    syslog.syslog(syslog.LOG_ERR, message)
+
+def is_process_running(process_name):
+    """Check if there is any running process that contains the given name."""
+    try:
+        # Use pgrep to check if the process is running
+        output = subprocess.check_output(["pgrep", "-f", process_name])
+        return bool(output.strip())
+    except subprocess.CalledProcessError:
+        return False
+
+def is_arp_update_stuck(timeout=TIMEOUT):
+    """Check if arp_update is stuck by looking for a ping command running longer than the specified timeout. (Default 1sec)"""
+    try:
+        arp_update_pid = subprocess.check_output(["pgrep", "-x", "arp_update"]).strip().decode('utf-8')
+        child_pids = subprocess.check_output(["pgrep", "-P", arp_update_pid]).strip().decode('utf-8').split()
+        for pid in child_pids:
+            cmd = subprocess.check_output(["ps", "-p", pid, "-o", "comm="]).strip().decode('utf-8')
+
+            if "ping" in cmd:
+                start_time = subprocess.check_output(["ps", "-p", pid, "-o", "etime="]).strip().decode('utf-8')
```

Comment:
if you use "etimes" instead of "etime" this would give you the value in seconds and you can get rid of the parse_elapsed_time function

Reply:
updated in the latest revision